spark/src/main/java/org/apache/iceberg/spark/SparkTableUtil.java

```diff
@@ -105,6 +105,20 @@
   private SparkTableUtil() {
   }
 
+  /**
+   * From Apache Spark
+   *
+   * Convert URI to String.
+   * Since URI.toString does not decode the uri, e.g. change '%25' to '%'.
+   * Here we create a hadoop Path with the given URI, and rely on Path.toString
+   * to decode the uri
+   * @param uri the URI of the path
+   * @return the String of the path
+   */
+  public static String uriToString(URI uri) {
```

Comment:
If this is going to be public, I'd rather find a better place than in `SparkTableUtil`, like `org.apache.iceberg.hadoop.Util`. I'm also fine with this being a private method here.

Reply:
I'll move it to Util, since this will probably come up again in the future an possibly in a non Spark Context (pretty sure all HadoopFS interactions will run into this if we ever implement methods for another system)